rule.proto

```diff
@@ -0,0 +1,221 @@
+syntax = "proto2";
+package dv360_excel_plugin;
+
+/* A message that represents a scheduled rule for manipulating DV360 entities.
+ */
+message Rule {
+  /* required
+   * The name of the rule, set by the user.
+   */
+  optional string name = 1;
+
+  /* required
+   * The id that uniquely identified each rule.
+   */
+  optional int32 id = 2;
+
+  // required
+  optional Action action = 3;
+
+  // required
+  optional Condition condition = 4;
+
+  // required
+  optional Schedule schedule = 5;
+}
+
+/*
+ * A message representing the action applied to a DV360 entity.
+ */
+message Action {
+  enum Type {
+    UNSPECIFIED_TYPE = 0;
+    CHANGE_LINE_ITEM_STATUS = 1;
+    CHANGE_BIDDING_STRATEGY = 2;
+    CHANGE_FIXED_BID_PRICE = 3;
+    DUPLICATE_LINE_ITEM = 4;
+  }
+
+  /* required
+   * The type of action.
+   */
+  optional Type type = 1;
+
+  /*
+   * The action parameters to change line items' status.
+   */
+  message ChangeLineItemStatusParams {
```

Comment:
probably you need advertiser id as well. check the API user guide, the request requires advertiser id.

Reply:
Whoops, forgot about it!